Content.Shared/SpittableContainer/SharedSpittableContainerSystem.cs

```diff
@@ -0,0 +1,134 @@
+using Content.Shared.Actions;
+using Content.Shared.DoAfter;
+using Content.Shared.IdentityManagement;
+using Content.Shared.Item;
+using Content.Shared.Popups;
+using Content.Shared.SpittableContainer.Components;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Containers;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.SpittableContainer;
+
+public abstract class SharedSpittableContainerSystem : EntitySystem
+{
+    [Dependency] private readonly SharedAudioSystem _audioSystem = default!;
+    [Dependency] private readonly SharedDoAfterSystem _doAfterSystem = default!;
+    [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
+    [Dependency] private readonly SharedActionsSystem _actionsSystem = default!;
+    [Dependency] protected readonly SharedContainerSystem _containerSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<SpittableContainerComponent, ComponentInit>(OnInit);
+        SubscribeLocalEvent<SpittableContainerComponent, ComponentShutdown>(OnShutdown);
+        SubscribeLocalEvent<SpittableContainerComponent, SwallowToContainerActionEvent>(OnSwallowToContainerAction);
+        SubscribeLocalEvent<SpittableContainerComponent, SwallowDoAfterEvent>(OnSwallowDoAfter);
+        SubscribeLocalEvent<SpittableContainerComponent, SpitFromContainerActionEvent>(OnSpitFromContainerAction);
+        SubscribeLocalEvent<SpittableContainerComponent, SpitFromContainerDoAfterEvent>(OnSpitDoAfter);
+    }
+
+    private void OnInit(Entity<SpittableContainerComponent> ent, ref ComponentInit args)
+    {
+        ent.Comp.Container = _containerSystem.EnsureContainer<Container>(ent.Owner, ent.Comp.Storage);
+
+        if (ent.Comp.SwallowActionPrototype != null)
+        {
+            EntityUid? SwallowActionUid = null;
+            _actionsSystem.AddAction(ent.Owner, ref SwallowActionUid, ent.Comp.SwallowActionPrototype);
+            if (SwallowActionUid != null)
+                ent.Comp.SwallowActionEntity = SwallowActionUid.Value;
+        }
+
+        if (ent.Comp.SpitContainerActionPrototype != null)
+        {
+            EntityUid? SpitActionUid = null;
+            _actionsSystem.AddAction(ent.Owner, ref SpitActionUid, ent.Comp.SpitContainerActionPrototype);
+            if (SpitActionUid != null)
+                ent.Comp.SwallowActionEntity = SpitActionUid.Value;
+        }
```

Comment:
You may try to create new general method with universal processing to optimise it.
Like this for example:
```C#
private void AddActionIfNeeded(EntityUid ownerEntity, ref EntityUid? actionEntity, string? actionPrototype)
{
    if (actionPrototype == null)
        return;

    EntityUid? actionUid = null;
    _actionsSystem.AddAction(ownerEntity, ref actionUid, actionPrototype);
    if (actionUid != null)
        actionEntity = actionUid.Value;
}
```
And then you can use it like this:
```C#
AddActionIfNeeded(ent.Owner, ref ent.Comp.SwallowActionEntity, ent.Comp.SwallowActionPrototype);
AddActionIfNeeded(ent.Owner, ref ent.Comp.SwallowActionEntity, ent.Comp.SpitContainerActionPrototype);
```